polling/infrequent/test_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'temporalio/activity'
+
+# A mock external service with simulated errors.
+module TestService
+  class TestServiceError < StandardError; end
+
+  @attempts = Hash.new(0)
+  ERROR_ATTEMPTS = 5
+
+  ComposeGreetingInput = Struct.new(:greeting, :name)
```

Comment:
Why have a struct for sending to activity if it just becomes a hash on receipt? Either just use hash everywhere, or add `require 'json/add/struct'` per https://docs.ruby-lang.org/en/master/JSON.html#module-JSON-label-Built-in+Additions so this can be a struct everywhere. I have no preference.

Reply:
remove the struct